form/persistence/persistence.cpp

```diff
@@ -9,71 +9,143 @@ using namespace form::detail::experimental;
 
 // Factory function implementation
 namespace form::detail::experimental {
-  std::unique_ptr<IPersistence> createPersistence()
+  std::unique_ptr<IPersistence> createPersistence(
+    const form::experimental::config::parse_config& config) //factory takes form config
   {
-    return std::unique_ptr<IPersistence>(new Persistence());
+    return std::unique_ptr<IPersistence>(new Persistence(config));
   }
 } // namespace form::detail::experimental
 
-Persistence::Persistence() : m_store(nullptr) { m_store = createStorage(); }
+Persistence::Persistence(const form::experimental::config::parse_config& config) :
+  m_store(nullptr), m_config(config) // constructor takes form config
+{
+  m_store = createStorage();
+}
 
 void Persistence::createContainers(const std::string& creator,
                                    const std::map<std::string, std::string>& products)
 {
   std::map<std::unique_ptr<Placement>, std::string> containers;
   for (const auto& [label, type] : products) {
-    containers.insert(std::make_pair(getPlacement(creator + "/" + label), type));
+    containers.insert(std::make_pair(getPlacement(creator, label), type));
   }
-  containers.insert(std::make_pair(getPlacement(creator + "/index"), "std::string"));
+  containers.insert(std::make_pair(getPlacement(creator, "/index"), "std::string"));
   m_store->createContainers(containers);
   return;
 }
 
-void Persistence::registerWrite(const std::string& label, const void* data, const std::string& type)
+void Persistence::registerWrite(const std::string& creator,
+                                const std::string& label,
+                                const void* data,
+                                const std::string& type)
 {
-  std::unique_ptr<Placement> plcmnt = getPlacement(label);
+  std::unique_ptr<Placement> plcmnt = getPlacement(creator, label);
   m_store->fillContainer(*plcmnt, data, type);
   return;
 }
 
 void Persistence::commitOutput(const std::string& creator, const std::string& id)
 {
-  std::unique_ptr<Placement> plcmnt = getPlacement(creator + "/index");
+  std::unique_ptr<Placement> plcmnt = getPlacement(creator, "/index");
```

Comment:
You could have decided here not to include the '/', since its no longer needed as separator ;)